### PR TITLE
DocumentationComment: Correct docstring parameters

### PR DESCRIPTION
**Pull request by adamgrieger**
Remove parameter descriptions for language and docstyle.
Add parameter description for docstyle_definition.

Closes https://github.com/coala/coala/issues/3998

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [X] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [X] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by supergr35** [comment]
ack bd75854

**Comment by supergr35** [comment]
unack bd75854

**Comment by supergr35** [comment]
Everything here looks good except in your commit shortlog, `DocumentationComment` -> `DocumentationComment.py` so it references the file. 

If this is your first PR, really good job. It looks like you followed our documentation really well 👍

**Comment by meetmangukiya** [comment]
@supergr5 do we enforce use of extensions in shortlog ? :/ I remember myself not using it sometimes :sweat_smile:

File: coalib/bearlib/languages/documentation/DocumentationComment.py [hunk 1]
```diff
@@ -20,14 +20,14 @@ def __init__(self, documentation, docstyle_definition,
...
-        :param language:      The language of the documention.
-        :param docstyle:      The docstyle used in the documentation.
-        :param indent:        The string of indentation used in front
-                              of the first marker of the documentation.
-        :param marker:        The three-element tuple with marker strings,
-                              that identified this documentation comment.
-        :param range:         The position range of type TextRange.
+        :param documentation:       The documentation text.
+        :param docstyle_definition: The docstyle_definition used in the
+                                    definition.
```
Comment (Udayan12167) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: Should be documentation right?

**Comment by Udayan12167** [comment]
@meetmangukiya We don't.

File: coalib/bearlib/languages/documentation/DocumentationComment.py (hunk 1 above)
Comment (userzimmermann) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: And I would not just repeat the identifier `docstyle_definition` in the description. A bit more detail about the expected data would be helpful.

**Comment by Asnelchristian** [comment]
this looks good to me

**Comment by Asnelchristian** [comment]
ack https://github.com/coala/coala/pull/4015/commits/45fa3b207a0491c5d3b0d83e308e9d527b790545

File: coalib/bearlib/languages/documentation/DocumentationComment.py [hunk 2]
```diff
@@ -20,14 +20,19 @@ def __init__(self, documentation, docstyle_definition,
...
+            The ``DocstyleDefinition`` instance that defines what docstyle is
+            being used in the documentation.
+        :param indent:
+            The string of indentation used in front of the first marker of the
+            documentation.
+        :param marker:
+            The three-element tuple with marker strings, that identified this
+            documentation comment.
+        :param range:
+            The position range of type TextRange.
```
Comment (userzimmermann) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: Just quickly put `TextRange` also in double back ticks like `DocstyleDefinition` and we are good to merge this :)

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4015/commits/994bff043d6c87618485c51505bf0590cb549b30

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4015/commits/994bff043d6c87618485c51505bf0590cb549b30

**Comment by userzimmermann** [comment]
Thanks @adamgrieger :)

**Comment by userzimmermann** [comment]
@rultor merge

**Comment by userzimmermann** [comment]
@adamgrieger Oh no! :D Please rebase

**Comment by Nosferatul** [comment]
ack 6f84b42

**Comment by Nosferatul** [comment]
@rultor merge

**Pull request merged**